pkg/controller/servicebindingrequest/sbrcontroller.go

```diff
@@ -113,18 +113,32 @@ func updateFunc(logger *log.Log) func(updateEvent event.UpdateEvent) bool {
 		isConfigMap := isOfKind(e.ObjectNew, "ConfigMap")
 
 		if isSecret || isConfigMap {
-			dataFieldsAreEqual, err := compareObjectFields(e.ObjectNew, e.ObjectOld, "data")
+			dataFieldsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "data")
 			if err != nil {
 				logger.Error(err, "error comparing object fields: %s", err.Error())
 				return false
 			}
-
 			logger.Debug("Predicate evaluated", "dataFieldsAreEqual", dataFieldsAreEqual)
 			return !dataFieldsAreEqual
 		}
+		logger.Debug("updateFunc", "Predicate evaluated for Kind", e.ObjectOld.GetObjectKind)
+		specsAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "spec")
+		if err != nil {
+			logger.Error(err, "error comparing object's spec fields: %s", err.Error())
+		}
+		logger.Debug("Predicate evaluated", "Specs of the Object in predicate filtering changed =", specsAreEqual)
 
-		// ignore updates to CR status in which case metadata.Generation does not change
-		return e.MetaOld.GetGeneration() != e.MetaNew.GetGeneration()
+		statusAreEqual, err := compareObjectFields(e.ObjectOld, e.ObjectNew, "status")
+		if err != nil {
+			logger.Error(err, "error comparing object's status fields: %s", err.Error())
+		}
+		logger.Debug("Predicate evalated", "Status of the Object in predicate filtering changed =", statusAreEqual)
+		if !specsAreEqual || !statusAreEqual {
+			logger.Info("Reconcile triggered")
```

Comment:
I think we can drop this line if you incorporate the above suggestion.

Reply:
Removed https://github.com/redhat-developer/service-binding-operator/pull/597/commits/b904a2186b93b06ef06213f2a23139572628abdc